### PR TITLE
太陽フレア : pylint対応

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/flareformer/main.py [hunk 1]
```diff
@@ -2,29 +2,28 @@
 
 import json
 import argparse
-import torch
-import structure.structure
-import torch.nn as nn
-import numpy as np
-import colored_traceback.always
```
Comment (ghost) [inline src/flareformer/main.py:9]: `colored_traceback.always`は消してはいけないので、戻しておいてください。
（importしてあることに意味のあるモジュールです。）

File: src/flareformer/dataloader/sampler.py [hunk 2]
```diff
@@ -1,34 +1,34 @@
-from torch.utils.data.sampler import BatchSampler
-from torch.utils.data import DataLoader
+"""Sampler for Flare Transformer"""
```
Comment (ghost) [inline src/flareformer/dataloader/sampler.py]: Flareformerが正式名称です。水平展開をお願いします。

File: src/flareformer/dataloader/sampler.py [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
     """
-
     def __init__(self, dataset: FlareDataset, n_classes: int, n_samples: int):
         print("Prepare Batch Sampler ...")
         loader = DataLoader(dataset)
         self.labels_list = []
-        for x, y, idx in tqdm(loader):
-            self.labels_list.append(np.argmax(y))
+        for _x, _y, idx in tqdm(loader):
+            self.labels_list.append(np.argmax(_y))
```
Comment (ghost) [inline src/flareformer/dataloader/sampler.py]: `_`をつけて回避するのは間違いです。名前を長くする方向に修正しましょう。

File: src/flareformer/utils/logs.py [hunk 4]
```diff
@@ -1,28 +1,32 @@
-import wandb as wandb_runner
+""" Scripts for logs """
```
Comment (ghost) [inline src/flareformer/utils/logs.py]: モジュールコメントなのに、scripts(複数形もおかしいですね)なのは違和感があります。
また、scriptsとsrcは根本的に格納するものが違うというお話もしたかと思いますので、
src配下にあるpythonコードのコメントに安易にscriptsという単語を使うのは良くないです。

Comment (ghost) [reply src/flareformer/utils/logs.py]: 水平展開をお願いします。

File: src/flareformer/dataloader/sampler.py (hunk 3 above)
Comment (ghost) [reply src/flareformer/dataloader/sampler.py]: 水平展開をお願いします。

**Review by ghost — CHANGES_REQUESTED** [review]
修正をお願いします。

File: src/flareformer/main.py [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
-
 from dataloader.dataloader import prepare_dataloaders
+from dataloader.flare import OneshotDataset
+from engine import train_epoch, eval_epoch
 
 
 def parse_params(dump: bool = False) -> Tuple[Namespace, Dict[str, Any]]:
+    """
+    Parse params
+    """
```
Comment (ghost) [inline src/flareformer/main.py]: コマンドライン引数をパース

File: src/flareformer/main.py [hunk 6]
```diff
@@ -140,6 +143,9 @@ def test(self):
         print(test_score)
 
     def predict_one_shot(self, imgs: torch.Tensor, feats: np.ndarray):
+        """
+        Predict oneshot
+        """
```
Comment (ghost) [inline src/flareformer/main.py]: 一発打ちを実行

（漢字表記も間違えないようにしてください。一発うちではだめということです。）

File: src/flareformer/preprocess/datasets.py [hunk 7]
```diff
@@ -1,24 +1,33 @@
...
+    Year section class
+    """
     start: int
     end: int
 
 
 def read_jsonl(path: str) -> List[dict]:
+    """
+    Read jsonl
+    """
```
Comment (ghost) [inline src/flareformer/preprocess/datasets.py]: josnlファイルをdictの配列として読み込む

File: src/flareformer/preprocess/datasets.py [hunk 8]
```diff
@@ -32,13 +41,19 @@ def detect_year_sections(jsonl: List[dict]):
 
 
 def split_dataset(jsonl: List[dict], sections: dict) -> dict:
+    """
+    Split dataset
+    """
```
Comment (ghost) [inline src/flareformer/preprocess/datasets.py]: dictの配列データを純粋な辞書形式に変換

File: src/flareformer/preprocess/datasets.py [hunk 9]
```diff
@@ -32,13 +41,19 @@ def detect_year_sections(jsonl: List[dict]):
...
     results = {}
     for k, v in sections.items():
         results[k] = jsonl[v.start:v.end + 1]
     return results
 
 
 def get_image(img_path: str, resize_size: int = 512) -> ndarray:
+    """
+    Get image
+    """
```
Comment (ghost) [inline src/flareformer/preprocess/datasets.py]: 指定されたパスの画像ファイルをnumpy形式で読み込む

File: src/flareformer/main.py [hunk 10]
```diff
@@ -159,7 +165,7 @@ def print_summary(self):
 
     def freeze_feature_extractor(self):
         """
-        Freeze feature extractor
+        抽出した特徴量を固定する
         """
```
Comment (ghost) [inline src/flareformer/main.py]: Feature Extractorの重みを固定する

Comment (ghost) [reply src/flareformer/main.py]: 必要であれば水平展開をお願いします。

File: src/flareformer/structure/attn.py [hunk 11]
```diff
@@ -1,29 +1,28 @@
-import torch
-import torch.nn as nn
-import torch.nn.functional as F
-
-import numpy as np
-
+"""学習に用いるAttentionモデルを定義するモジュール"""
```
Comment (ghost) [inline src/flareformer/structure/attn.py]: `学習に用いる`について、学習だけではないような気がします。
（学習に用いると書かなくても良いかも？）

File: src/flareformer/structure/attn.py [hunk 12]
```diff
@@ -214,6 +217,7 @@ def forward(self, queries: Tensor,
 
 
 class TriangularCausalMask():
+    """Triangular Causal Masking を設定するクラス"""
```
Comment (ghost) [inline src/flareformer/structure/attn.py]: `ing`はなぜついているのでしょうか。
また、設定はやめましょう。ここはどちらかといえば生成ではないでしょうか。

Comment (ghost) [reply src/flareformer/structure/attn.py]: ProbMaskも同様

File: src/flareformer/structure/attn.py [hunk 13]
```diff
@@ -222,10 +226,12 @@ def __init__(self, B, L, device="cpu"):
 
     @property
     def mask(self):
+        """mask関数"""
```
Comment (ghost) [inline src/flareformer/structure/attn.py]: maskを返す

File: src/flareformer/structure/misc.py [hunk 14]
```diff
@@ -1,3 +1,4 @@
+"""その他の雑多なモデルを定義するモジュール"""
```
Comment (ghost) [inline src/flareformer/structure/misc.py]: その他の雑多**な**
ではなく、
その他雑多**に**
にしていただけますか。

File: src/flareformer/structure/structure.py [hunk 15]
```diff
@@ -84,6 +93,9 @@ def forward(self, img_list: Tensor, feat: Tensor) -> Tuple[Tensor, Tensor]:
         return output, x
 
     def freeze_feature_extractor(self):
+        """
+        抽出した特徴量を固定する
+        """
```
Comment (ghost) [inline src/flareformer/structure/structure.py]: 前の指摘と同様

File: src/flareformer/structure/structure.py [hunk 16]
```diff
@@ -195,12 +210,18 @@ def _init_weights(self, m: Union[nn.Module, nn.Module, nn.Module]):
             nn.init.constant_(m.bias, 0)
 
     def forward_features(self, x: Tensor) -> Tensor:
+        """
+        順伝播で用いる特徴量を更新する
+        """
```
Comment (ghost) [inline src/flareformer/structure/structure.py]: 要検討
`特徴量を更新`という考え方はないので

File: src/flareformer/structure/structure.py [hunk 17]
```diff
@@ -236,13 +260,16 @@ def forward(self, x: Tensor) -> Tensor:
 
 
 class InformerEncoderLayer(nn.Module):
+    """
+    Informer encoder layer
```
Comment (ghost) [inline src/flareformer/structure/structure.py]: 単語の頭を大文字にしてください。

File: src/flareformer/utils/server.py [hunk 18]
```diff
@@ -2,33 +2,35 @@
...
     month: str
     day: str
     hour: str
 
 
 class CallbackServer:
+    """ Callback server class """
     @staticmethod
     def get_tensor_image(img_buff):
+        """ Get tensor image """
```
Comment (ghost) [inline src/flareformer/utils/server.py]: 画像のRAWデータをTensorに変換

File: src/flareformer/utils/server.py [hunk 19]
```diff
@@ -2,33 +2,35 @@
...
 class Date(BaseModel):
+    """ Date class """
     year: str
     month: str
     day: str
     hour: str
 
 
 class CallbackServer:
+    """ Callback server class """
```
Comment (ghost) [inline src/flareformer/utils/server.py]: 一発打ち・画像取得のためのコールバックを定義し、サーバを起動するクラス

File: src/flareformer/utils/server.py [hunk 20]
```diff
@@ -37,6 +39,7 @@ def get_tensor_image(img_buff):
 
     @staticmethod
     def get_tensor_image_from_path(path):
+        """ Get tensor image from path """
```
Comment (ghost) [inline src/flareformer/utils/server.py]: 指定されたパスの画像ファイルを読み込んでTensor形式で返す

File: src/flareformer/utils/utils.py [hunk 21]
```diff
@@ -1,10 +1,10 @@
-import numpy as np
-import matplotlib.pyplot as plt
-import torch
+"""学習に用いる設定値を決めるモジュール"""
```
Comment (ghost) [inline src/flareformer/utils/utils.py]: 学習向けのユーティリティモジュール

**Review by ghost — CHANGES_REQUESTED** [review]
修正をお願いします。

File: src/flareformer/structure/structure.py (hunk 16 above)
Comment (ghost) [reply src/flareformer/structure/structure.py]: 全結合層に入力する特徴量までの順伝播を定義する関数
にします

File: src/flareformer/preprocess/utils.py [hunk 22]
```diff
@@ -8,7 +9,11 @@ def make_prefix(output_path: str, year: int, data_type: str) -> str:
     return f"{output_path}/data_{year}_{data_type}"
 
 
-def identity(x): return x
+def identity(_x):
+    """
+    Identity
+    """
+    return _x
```
Comment (ghost) [inline src/flareformer/preprocess/utils.py]: これは`_`なくても行けるような気がするのですが、
pylintに指摘されるのでしょうか。

File: src/flareformer/dataloader/sampler.py [hunk 23]
```diff
@@ -1,24 +1,24 @@
...
     BatchSampler - from a MNIST-like dataset, samples n_classes and within these classes samples n_samples.
     Returns batches of size n_classes * n_samples
     """
-
     def __init__(self, dataset: FlareDataset, n_classes: int, n_samples: int):
         print("Prepare Batch Sampler ...")
         loader = DataLoader(dataset)
         self.labels_list = []
-        for x, y, idx in tqdm(loader):
+        for _x, y, idx in tqdm(loader):
```
Comment (ghost) [inline src/flareformer/dataloader/sampler.py]: `_x`ではなくて、`_`？

File: src/flareformer/structure/attn.py [hunk 24]
```diff
@@ -222,10 +226,12 @@ def __init__(self, B, L, device="cpu"):
 
     @property
     def mask(self):
+        """maskを返す"""
         return self._mask
 
 
 class ProbMask():
+    """Probabilistic Maskを生成するクラス"""
```
Comment (ghost) [inline src/flareformer/structure/attn.py]: `Prob Attention`と揃えて、`Prob Mask`で良いと思います。
（論文にそう書いてあったのであればOKですが）

**Review by ghost — CHANGES_REQUESTED** [review]
修正をお願いします。

File: src/flareformer/preprocess/utils.py (hunk 22 above)
Comment (ghost) [reply src/flareformer/preprocess/utils.py]: pylintで指摘されてしまうのですが、これまでの指示通り使用している変数なので`_`は使用しないようにしようと思います。

File: src/flareformer/structure/attn.py (hunk 24 above)
Comment (ghost) [reply src/flareformer/structure/attn.py]: 自分用に付け加えた感はあります。承知しました。

File: src/flareformer/preprocess/utils.py (hunk 22 above)
Comment (ghost) [reply src/flareformer/preprocess/utils.py]: in_x
にいたしました

**Review by ghost — APPROVED** [review]
確認しました。OKです。
